src/Forms/ValidationList/index.js

```diff
@@ -0,0 +1,135 @@
+import { identity, flow, isEmpty, isEqual } from 'lodash/fp';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getActiveErrors, getErrorMessages } from './service';
+import styles from './index.scss';
+import withStyles from '../../../util/withStyles';
+
+class ValidationList extends Component {
+  static propTypes = {
+    data: PropTypes.shape({
+      values: PropTypes.object
+    }),
+    errors: PropTypes.object
+  };
+
+  static defaultProps = {
+    translateFn: identity
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.ERROR_DELAY = 500;
+    this.INTERACTION_THRESHOLD = 0;
+
+    this.state = {
+      isVisible: false,
+      timeoutId: null,
+      lastInteractionTime: null
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
This component has quite a lot of changes for the Brazil shop. Just to keep that in mind.